lightning/src/ln/channel.rs

```diff
@@ -5460,6 +5461,8 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 			// Read the 8 bytes of backwards-compatibility ChannelConfig data.
 			let mut _val: u64 = Readable::read(reader)?;
 		}
+		
+		handshake_config.as_mut().unwrap().commit_upfront_shutdown_pubkey = Readable::read(reader)?;
```

Comment:
We can't just add a new field to be read in the middle of the object - its not there in old serialized data! Instead, we should, I think, just store the `commit_upfront_shutdown_pubkey` field explicitly in a new boolean and load it explicitly.